### DISCUSSION TITLE
It is not possible to use the same events between two aggregates.

### DISCUSSION THREAD

**Discussion by unixslayer** [ecotoneframework/ecotone-dev#272] [Ideas]
Consider following scenario: The `Cycle` aggregate has the certificate issuing logic. `Certificate` is an aggregate with functionalities independent of `Cycle`, such as suspension.

It is not possible to use the same events between aggregates when the `Certificate` aggregate factory method is an `EventHandler` returning the event originally returned by `Cycle`. Both aggregates are saved correctly, but after saving the `Certificate`, the published `CertificateIssued` event causes a loop and triggers the factory method again for the same parameters.

Proper PR with test case will be introduced.

**Comment by dgafka** [comment]
This is expected behaviour, as events may trigger aggregate that have published them. 
So what do we want to do with this issue? :)

**Comment by unixslayer** [comment]
@dgafka My original idea was to use the same event for more than one stream. I have a business case that justifies this behavior. Looking for a similar solution I found that EventStoreDB gives the possibility to emit events by linking. However, the current implementation of `EventStreamEmitter` won't add proper metadata to events being linked to another stream. 

I would like to keep this issue open for a while and try to come up with a proper solution.
